MyCryptoHQ/MyCrypto
# Handle Gas / Estimates on a Per Network Basis

Closes #1116 

### Description

This doesn't solve every issue outlined in #1116, I figured some of those were bridges to cross when we came to it. But it does have each network declare some gas information.

### Changes

* Adds `gasPriceSettings` and `shouldEstimateGasPrice` configurations to each static network.
* Saga no longer requests gas price if `network.shouldEstimateGasPrice` is false.
* Saga uses network's default gas settings if request fails, or it shouldn't estimate the gas price
* Adds chainId as an attribute to gasEstimates so that cached estimates aren't re-used for a different network
* Reworks gas slider component code a bit to rely more on the gas "estimate"
* Reworks gas slider input handler code to allow for decimal values from the slider (ETC is 0.1gwei - 10gwei, so it steps in 0.1 increments)
* Refactored `staticNetwork.spec.ts` tests to be less copy-paste
* Removed the unused GasPriceDropdown from header

### TODO / Solve Later / Needs Feedback

* Figure out what multiple network gas estimates would look like
    * Maybe instead of shouldEstimateGasPrice, we have a gasPriceEstimateURL or something like that. We can do this in a separate PR if we get an estimator on another network.
* Confirm if gas price defaults are good for other networks.
    * I just looked at the explorers, and took some averages. UBIQ seemed weirdly high, everything was around 40gwei. Is that v3's fault for defaulting there?
* I didn't do anything to handle no-fee networks. I think we should handle that in a separate PR if we add one as a static network.
* Not confident in my nomenclature, feel free to suggest better names for things.

### Steps to Test

1. Run app with branch `network-gas-behavior`
2. Unlock a wallet on /send on the ETH network, confirm gas is estimated via network request
3. Switch to ETC network
4. Unlock a wallet on /send, confirm that range and initial value is different, no network request made
5. Slide it around and make sure it's not screwed up

## Screenshots

### Fraction gas price on ETC

![screen shot 2018-02-21 at 6 03 40 pm](https://user-images.githubusercontent.com/649992/36510607-9666de38-1731-11e8-89c1-84a86e935f18.png)
